src/response.rs

```diff
@@ -151,15 +151,15 @@ impl ResponseHeaders {
         self.write_allow_header(buf)?;
         self.write_deprecation_header(buf)?;
 
-        if self.content_length != 0 {
+        if let Some(content_length) = self.content_length {
```

Comment:
I think it would be nicer if we keep `content_length` just a `u32` and have this condition `if status != 204` (technically should be "if status not in [204, 1xx]", but we don't support 1xx status). That way, micro-http enforces the `Content-Length` related parts of RFC9110 instead of leaving it up to the caller to get it right (because if I understand correctly, right now, whether we add the content-length header depends on .body() being called, meaning we could have a 204 response with content length header, and a 200 without depending on whether we call .body() or not). It'd also be nice to disallow calling .body() if the status is set to 204 NO CONTENT, but that's out of scope for this PR

Reply:
I think centralizing that logic here would be nice, and we can add comments to refer to the RFCs.

Reply:
yeah, your understanding is correct. In my implementation, the presence of `Content-Length` field depends on whether `set_body()` was called or not. But IMHO, whether `Content-Length` must not be contained depends on what request the response was for. micro-http doesn't recognize such context when creating a response and it leaves room for users to decide what response to create. It's going to be a bit complex to implement it fully complying with RFC. At the moment, micro-http doesn't support many stuff, but I'm a bit cautious when adding a new stuff to micro-http. wdyt?

https://datatracker.ietf.org/doc/html/rfc9110#name-content-length
> A server MAY send a Content-Length header field in a response to a HEAD request ([Section 9.3.2](https://datatracker.ietf.org/doc/html/rfc9110#HEAD)); a server MUST NOT send Content-Length in such a response unless its field value equals the decimal number of octets that would have been sent in the content of a response if the same request had used the GET method.
> 
> A server MAY send a Content-Length header field in a [304 (Not Modified)](https://datatracker.ietf.org/doc/html/rfc9110#status.304) response to a conditional GET request ([Section 15.4.5](https://datatracker.ietf.org/doc/html/rfc9110#status.304)); a server MUST NOT send Content-Length in such a response unless its field value equals the decimal number of octets that would have been sent in the content of a [200 (OK)](https://datatracker.ietf.org/doc/html/rfc9110#status.200) response to the same request.
> 
> A server MUST NOT send a Content-Length header field in any response with a status code of [1xx (Informational)](https://datatracker.ietf.org/doc/html/rfc9110#status.1xx) or [204 (No Content)](https://datatracker.ietf.org/doc/html/rfc9110#status.204). A server MUST NOT send a Content-Length header field in any [2xx (Successful)](https://datatracker.ietf.org/doc/html/rfc9110#status.2xx) response to a CONNECT request ([Section 9.3.6](https://datatracker.ietf.org/doc/html/rfc9110#CONNECT)).

Reply:
Discussing with Pablo and Patrick, we decided to conform with not everything of RFC in this PR. Instead, we'll alway set `Content-Length` header in non-204/1XX responses with a TODO comment for the non-conforming parts.